JoyboyCommunity/src/screens/Tips/index.tsx

```diff
@@ -62,7 +65,62 @@ export const Tips: React.FC = () => {
     });
     const balance = uint256.uint256ToBN({low: balanceLow, high: balanceHigh});
 
-    if (balance < fee) {
+    // Send the claim through the wallet
+
+    const event = await getNostrEvent(BigInt(0));
+
+    const signature = event.sig ?? '';
+    const signatureR = signature.slice(0, signature.length / 2);
+    const signatureS = signature.slice(signature.length / 2);
+
+    const claimCalldata = CallData.compile([
+      uint256.bnToUint256(`0x${event.pubkey}`),
+      event.created_at,
+      event.kind ?? 1,
+      byteArray.byteArrayFromString(JSON.stringify(event.tags)),
+      {
+        deposit_id: cairo.felt(depositId),
+        starknet_recipient: connectedAccount.address,
+        gas_token_address: ETH[CHAIN_ID].address,
+        gas_amount: uint256.bnToUint256(0),
+      },
+      {
+        r: uint256.bnToUint256(`0x${signatureR}`),
+        s: uint256.bnToUint256(`0x${signatureS}`),
+      },
+      uint256.bnToUint256(0),
+    ]);
+
+    let estimateFee;
+    try {
+      estimateFee = await connectedAccount?.account?.estimateInvokeFee({
+        contractAddress: ESCROW_ADDRESSES[await provider.getChainId()],
+        entrypoint: Entrypoint.CLAIM,
+        calldata: claimCalldata,
+      });
+    } catch (error) {
+      console.log("Can't estimate UI fees", error);
+    }
+
+    if (!estimateFee?.overall_fee) {
+      try {
+        nonce = await connectedAccount?.account?.getNonce();
+      } catch (error) {
+        console.log('Get nonce error => maybe first tx', error);
+      }
+    }
+
+    /** @TODO better check for backed call need */
+    // - if nonce is undefined (no tx send) and no estimate fee on the UI
+    // - Balance < estimate fee
+    // - Nonce is ok (First tx send) but no balance min
+    const isBackendNeeded =
+      (!nonce && !estimateFee) ||
+      (estimateFee?.overall_fee && balance < estimateFee?.overall_fee) ||
+      (nonce && Number(nonce) > 0 && balance <= minAmountBalanceToClaim);
```

Comment:
Check is too complex, only check needed is either the account has balance, or not.